src/schemas/microservice.js

```diff
@@ -45,11 +47,12 @@ const microserviceUpdate = {
     "volumeMappings": {
       "type": "array",
       "items": {"$ref": "volumeMappings"}},
-    "ports": {"$ref": "/ports"},
+    "ports": {
+      "type": "array",
+      "items": {"$ref": "ports"}},
```

Comment:
are you sure that here should be "$ref": "ports" ?
Subschema ports has id "/ports" right now.

Reply:
done